EIPS/eip-5089.md

```diff
@@ -0,0 +1,391 @@
+---
+eip: 5089
+title: Principal Token
+description: Principal tokens (zero-coupon tokens) are redeemable for a single underlying ERC-20 token at a future timestamp.
+author: Julian Traversa (@JTraversa), Robert Robbins (@robrobbins), Alberto Cuesta Cañada (@alcueca)
+discussions-to: https://ethereum-magicians.org/t/eip-5089-principal-token-standard/9259
+status: Draft
+type: Standards Track
+category: ERC
+created: 2022-05-01
+requires: 20, 2612
+---
+
+## Abstract
+
+The following specification is an API for principal tokens --
+representing ownership of an underlying [ERC-20](./eip-20.md) token at a future timestamp.
+This specification is an extension on the [ERC-20](./eip-20.md) token that provides basic functionality for depositing
+and withdrawing tokens and reading balances and the ERC-2612 specification that provides
+[EIP-712](./eip-712.md) signature based approvals.
+
+## Motivation
+
+Principal tokens have an increasing lack of standardization which has led to a difficult to navigate development space and diverse implementation
+schemes.
+
+The primary examples include yield tokenization platforms which strip future yield leaving a principal
+token behind, as well as fixed-rate money-markets which utilize principal tokens as a medium
+to lend/borrow.
+
+This inconsistency in implementation makes integration difficult at the application layer as well as
+wallet layer which are key catalysts for the space's growth. 
+Developers are currently expected to implement individual adapters for each principal token, as well as adapters for
+their pool contracts, and many times adapters for their custodial contracts as well, wasting significant developer resources. 
+
+## Specification
+
+All ERC-5089 Principal Tokens (PTs) MUST implement [ERC-20](./eip-20.md) to represent ownership of future underlying redemption.
+If a PT is to be non-transferrable, it MAY revert on calls to `transfer` or `transferFrom`.
+The [ERC-20](./eip-20.md) operations `balanceOf`, `transfer`, `totalSupply`, etc. operate on the Principal Token balance.
+
+All ERC-5089 Principal Tokens MUST implement [ERC-20](./eip-20.md)'s optional metadata extensions.
+The `name` and `symbol` functions SHOULD reflect the underlying token's `name` and `symbol` in some way, as well as the origination protocol, and in the case of yield tokenization protocols, the origination money-market.
+
+ERC-5089 tokenized Vaults MAY implement [EIP-2612](./eip-2612.md) to improve the UX of approving PTs on various integrations.
+
+### Definitions:
+
+- underlying: The token that Principal Tokens are redeemable for at maturity.
```

Comment:
Ideally this standard could be made as an extension over EIP-4626. The only addition here is really the `maturity` and the definition of slippage. Is there something I am missing?

Reply:
Added a comment to the effect in the Rationale, thanks.
```
This EIP draws heavily on the design of [EIP-4626](./eip-4626.md) because technically Principal Tokens could be described as a subset of Yield Bearing Vaults, extended with a `maturity` variable and restrictions on the implementation. However, extending [EIP-4626](./eip-4626.md) would force PT implementations to include methods (namely, `mint` and `deposit`) that are not necessary to the business case that PTs solve. It can also be argued that partial redemptions (implemented via `withdraw`) are rare for PTs.
```